test/library/standard/Socket/connectblocking.chpl

```diff
@@ -0,0 +1,148 @@
+use UnitTest;
+use Socket;
+
+proc test_connection_ipv4_dns(test: borrowed Test) throws {
+  var port:uint(16) = 8811;
+  var host = "127.0.0.1";
+  var address = ipAddr.ipv4(IPv4Localhost, port);
+  var server = listen(address);
+  begin {
```

Comment:
I would expect that any time you are using `begin` that you also enclose the code in a `sync` block. For these tests, it would be reasonable to put a `sync` block around the test body. (I don't remember if the unit test infrastructure always adds `sync` blocks, though).

Reply:
For this I think we don't actually need `sync` because the calls made in main thread are blocking and rely on `begin` block so it will not proceed further without completing which might be a good test as well because we don't want to proceed (similar to sync)

Reply:
Sure. But my concern is - what if something goes wrong? If we want to check that something in one task happen before something in another task, we can check that (e.g. with an atomic variable). But if something goes wrong and the blocking does not occur, will the test core-dump?